src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
The order should be 1) SettingCredentialProvider, 2) All plugin credential providers in a deterministic order, 3) proxy provider.

The plugin credential providers all check if they handle given URI and fall through to the next in the list if they don't.  Proxy provider does not do this and will always prompt for basic credentials, so it must be last.

The order of plugin providers is not specified, but it should be deterministic so that it always uses the same order so users don't see inconsistent behavior.